src/components/BotSetting/BotSetting.tsx

```diff
@@ -10,102 +10,145 @@ import { BotDescription } from './BotDescription'
-    usageLimitPerUser: 1,
-    usageLimitPerUserType: UsageLimitTypeEnum.PerDay,
-    userLimitWarningMsg: `You've reached the message limit.`,
-  }
+  const { data: sources, refetch: refetchBotSettings } =
+    api.bot.getById.useQuery(id as string)
 
   const form = useForm<BotSettingData>({
-    defaultValues: defaultValuesForm,
     resolver: zodResolver(schema),
+    mode: 'all',
   })
 
-  const onSubmit = async (data: typeof defaultValuesForm) => {
-    const dataFormat: BotSettingData = {
+  const {
+    handleSubmit,
+    reset,
+    setValue,
+    formState: { isSubmitting, isDirty },
+  } = form
+
+  const onSubmit = async (props: BotSettingData) => {
+    const payload: BotSettingData = {
       botId: id as string,
-      name: data.name,
-      description: data.description,
-      modelId: Number(data.modelId),
-      noSourceWarningMsg: data.noSourceWarningMsg,
-      serverErrorMsg: data.serverErrorMsg,
-      usageLimitPerUser: Number(data.usageLimitPerUser),
-      usageLimitPerUserType: Number(data.usageLimitPerUserType),
-      userLimitWarningMsg: data.userLimitWarningMsg,
+      name: props.name,
+      description: props.description,
+      noSourceWarningMsg: props.noSourceWarningMsg,
+      serverErrorMsg: props.serverErrorMsg,
+      userLimitWarningMsg: props.userLimitWarningMsg,
+      modelId: Number(props.modelId),
+      usageLimitPerUser: Number(props.usageLimitPerUser),
+      usageLimitPerUserType: Number(props.usageLimitPerUserType),
     }
 
     try {
-      updateBotSettings(dataFormat)
+      updateBotSettings(payload)
     } catch (error) {
       console.log(error)
     }
+
+    await refetchBotSettings()
   }
 
-  const { handleSubmit } = form
+  const resetData = useCallback(
+    (data?: typeof sources) => {
+      if (!data) return
+      reset({
+        botId: id as string,
+        name: data.name!,
+        description: data.description!,
+        noSourceWarningMsg: data.noSourceWarningMsg!,
+        serverErrorMsg: data.serverErrorMsg!,
+        userLimitWarningMsg: data.userLimitWarningMsg!,
+        modelId: Number(data.modelId),
+        usageLimitPerUser: Number(data.usageLimitPerUser!),
+        usageLimitPerUserType: Number(data.usageLimitPerUserType!),
+      })
+    },
+
+    [reset],
+  )
+
+  useEffect(() => {
+    if (sources && !isInitialData.current) {
+      isInitialData.current = true
+      setValue('name', sources.name!)
+      setValue('description', sources.description!)
+      setValue('noSourceWarningMsg', sources.noSourceWarningMsg!)
+      setValue('serverErrorMsg', sources.serverErrorMsg!)
+      setValue('userLimitWarningMsg', sources.userLimitWarningMsg!)
+      setValue('usageLimitPerUser', Number(sources.usageLimitPerUser))
+      setValue('modelId', Number(sources.modelId))
+      setValue('usageLimitPerUserType', Number(sources.usageLimitPerUserType))
+    }
+  }, [sources])
+
+  useEffect(() => {
+    resetData(sources)
+  }, [resetData, sources])
```

Comment:
Why do we need to do this?

Reply:
Because I think call refetchBotSettings() after submitting. The sources are updated, it puts that value into the form again.

Reply:
What happened if we don't do it? Isn't your submitted data the latest? I'm seeing this step is a redundance.